### ISSUE TITLE
Rename(refactor) on method's parameter not applying to method in super/sub class.

### ISSUE THREAD

**Issue by hamsbrar** [dart-lang/sdk#53187]
```dart
class T1 {
  void f(int a) {}
}

class T2 extends T1 {
  @override
  void f(int a) {}
}
```
Renaming `a` in `T1.f` doesn't rename `a` in `T2.f`.
```sh
> dart --version
Dart SDK version: 3.0.7 (stable) (Mon Jul 24 13:17:56 2023 +0000) on "linux_x64"

> code --list-extensions --show-versions | grep dart-code
Dart-Code.dart-code@3.68.1
```

I think rename `a` in `T1.f` should also rename `a` in `T2.f` and vice-versa. And if there are more types in hierarchy that has `f`, then it should be applied to those as well.

**Comment by scheglov** [comment]
Strongly speaking semantics does not require this, in contrast to named formal parameters, which we do rename.

But I agree that it would be reasonable to have an option to rename even positional formal parameters in the whole hierarchy, probably even making this the default behavior.

**Comment by bwilkerson** [comment]
I think it would be reasonable to use the enablement of `avoid_renaming_method_parameters` as the option that controls whether matching parameters in overridden / overriding method are also renamed. I wouldn't make it the default behavior, though.

**Comment by FMorschel** [comment]
> Oh there's one issue though, it'll be difficult for a user who wants to start using a different name for ith parameter in a single method.

@DanTup, I remember that when renaming things we do have a message for confirmation when the rename is shadowing something else, not sure how it works though. Could this be another question like that - if it will rename this parameter in more than one place ask if it should apply everywhere or only locally?

**Comment by DanTup** [comment]
>  if it will rename this parameter in more than one place ask if it should apply everywhere or only locally?

It's possible, but I think that would happen a lot, and having a toast notification (sometimes that users don't even notice because the bottom right corner of the editor might be far away from the code they're staring at) might not be a great experience.

That said, I'm also not sure I understand the scenario... If the lint is disabled, then the user can rename the local parameter only. If the lint is enabled, it will rename all. It seems unlikely a user wants to rename just the local parameter with the lint enabled, because that would produce a lint warning?

**Comment by FMorschel** [comment]
Yeah, you're probably right. But the other way around is what I would mind the most in relation to this. If you want to rename all references together but can't because of the lint being disabled. 

I guess you could say the user does want the lint here but this could very much be a one time thing (they'd like a single method three to care about this way). I don't think it would be that impactful to us though, we could simply say to activate the lint temporarily and rename then if that is what they want.

**Comment by FMorschel** [comment]
Here is a CL for this https://dart-review.googlesource.com/c/sdk/+/412520

**Comment by bwilkerson** [comment]
> ... we could simply say to activate the lint temporarily and rename then if that is what they want.

That isn't very discoverable.

**Comment by FMorschel** [comment]
Agreed, but this lint is on the recomended set https://dart.dev/tools/linter-rules/avoid_renaming_method_parameters. Not sure what else we could do here for this.
